src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java

```diff
@@ -345,7 +350,12 @@ private void cacheSchemaFetchError(
         schemaId.schemaId(),
         e.getMessage()
     ));
-    schemaErrors.writeSchemaIdByConnectionId(new SchemaErrors.ConnectionId(context.getConnectionId()), schemaId, errorMessage);
+    schemaErrors.writeSchemaIdByConnectionId(
+        context.getConnectionId(),
+        String.valueOf(schemaId.schemaId()),
```

Comment:
I think this was a miss on my part, but we shouldn't really be converting the schema IDs to strings and then back to integers. If we need them as integers, why don't we expect an integer parameter?